src/libs/polycube/src/utils.cpp

```diff
@@ -307,6 +311,49 @@ std::string get_netmask_from_CIDR(const int cidr) {
   return inet_ntoa(addr);
 }
 
+void split_ip_and_prefix(const std::string &ip_and_prefix,
+                        std::string &ip_address, std::string &netmask) {
+  // ip_and_prefix = ip_address/prefix
+  std::istringstream split(ip_and_prefix);
+  std::vector<std::string> info;
+  char split_char = '/';
+  for (std::string each; std::getline(split, each, split_char);
+       info.push_back(each))
+    ;
+  ip_address = info[0];
+  netmask = get_netmask_from_CIDR(std::atoi(info[1].c_str()));
+}
+
+bool address_in_subnet(const std::string &ip, const std::string &netmask,
+                       const std::string &network) {
+  uint32_t ipAddress = ip_string_to_be_uint(ip);
+  uint32_t mask = ip_string_to_be_uint(netmask);
+  uint32_t net = ip_string_to_be_uint(network);
+  if ((ipAddress & mask) == (net & mask))
+    return true;
+  else
+    return false;
+}
+
+std::string get_network_from_ip(const std::string &ip,
+                                const std::string &netmask) {
+  // get the network from ip
+  uint32_t address = ip_string_to_be_uint(ip);
```

Comment:
This comment is not related to this PR; however, I feel the name of this function is misleading. In fact, we're not transforming a string into a "big endian" number, but in "network byte order" number.
So, I would kindly ask you to open a new PR and rename the following functions (and add a comment in the code, so that people know that "nbo" is "network byte order"):
- ip_string_to_be_uint ==> ip_string_to_nbo_uint
- be_uint_to_ip_string ==> nbo_uint_to_ip_string
The same for the other functions that operate on MAC addresses.